wpilibNewCommands/src/main/native/cpp/frc2/command/DeferredCommand.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include "frc2/command/DeferredCommand.h"
+
+#include <wpi/sendable/SendableBuilder.h>
+
+#include "frc2/command/PrintCommand.h"
+
+using namespace frc2;
+
+DeferredCommand::DeferredCommand(wpi::unique_function<Command*()> supplier,
+                                 std::span<Subsystem* const> requirements)
+    : m_supplier{std::move(supplier)} {
+  AddRequirements(requirements);
+}
+
+DeferredCommand::DeferredCommand(wpi::unique_function<Command*()> supplier,
+                                 std::initializer_list<Subsystem*> requirements)
+    : m_supplier{std::move(supplier)} {
+  AddRequirements(requirements);
+}
+
+DeferredCommand::DeferredCommand(wpi::unique_function<CommandPtr()> supplier,
+                                 std::span<Subsystem* const> requirements)
+    : DeferredCommand(
+          [lambdaSupplier = std::move(supplier),
+           holder = std::optional<CommandPtr>{}]() mutable {
```

Comment:
I'm not sure the optional here is meaningful? (If CommandPtr has a default ctor, that is.) Open for discussion.

Reply:
I copied the impl from ProxyCommand in https://github.com/wpilibsuite/allwpilib/pull/5155/. I'm not sure how else to have a temporary "nothing" holder. We could set it to `Commands::None()`, maybe?